examples/HMC.ipynb

```diff
@@ -254,13 +221,24 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 11,
+   "execution_count": 9,
    "id": "6729c9d3",
    "metadata": {},
-   "outputs": [],
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/home/remi/.virtualenvs/aehmc-nuts/lib/python3.9/site-packages/aeppl/joint_logprob.py:135: UserWarning: Found a random variable that was neither among the observations nor the conditioned variables: normal_rv{0, (0, 0), floatX, False}(RandomGeneratorSharedVariable(<Generator(PCG64) at 0x7F8C4015AAC0>), TensorConstant{[]}, TensorConstant{11}, InplaceDimShuffle{x}.0, InplaceDimShuffle{x}.0)\n",
+      "  warnings.warn(\n",
+      "/home/remi/.virtualenvs/aehmc-nuts/lib/python3.9/site-packages/aeppl/utils.py:249: UserWarning: No value variable found for normal_rv{0, (0, 0), floatX, False}.out; the random variable will not be replaced.\n",
+      "  warnings.warn(\n"
```

Comment:
These warnings are scary, because they could mean that a `RandomVariable` is present in the log-likelihood graph and will be sampled.

I ran this example locally and didn't get these warnings, so it might just be an artifact of an old evaluation, but it's scary nonetheless!

If you ever want to make sure there isn't any sampling in your log-likelihood, you can run something like the following:
```python
from aesara.tensor.random.op import RandomVariable


assert not any(
    isinstance(v.owner.op, RandomVariable) for v in logprob_fn.maker.fgraph.variables if v.owner
)

```

The graph given by `logprob_fn.maker.fgraph` is optimized, which should get rid of any unnecessary `RandomVariable` nodes.  For example, `Y_rv.shape` will produce a `Shape` node with `Y_rv` as the input, but the shape inference that's run during optimization will replace it with a more direct computation of the shape that doesn't involve sampling `Y_rv`.

Reply:
Yes this is from a previous run and left from a merge conflict that I didn't clean carefully enough. I just re-ran the notebook and they disappeared.